Networking/NetworkingTests/Remote/CouponsRemoteTests.swift

```diff
@@ -350,6 +350,52 @@ final class CouponsRemoteTests: XCTestCase {
         let resultError = try XCTUnwrap(result.failure as? NetworkError)
         XCTAssertEqual(resultError, .unacceptableStatusCode(statusCode: 500))
     }
+
+    // MARK: - Retrieve coupon
+
+    /// Verifies that retrieveCoupon properly parses the `coupon` sample response.
+    ///
+    func test_retrieveCoupon_properly_returns_parsed_coupon() throws {
+        // Given
+        let sampleCouponID: Int64 = 720
+        let remote = CouponsRemote(network: network)
+        network.simulateResponse(requestUrlSuffix: "coupons/\(sampleCouponID)", filename: "coupon")
+
+        // When
+        let result = waitFor { promise in
+            remote.retrieveCoupon(for: self.sampleSiteID, couponID: sampleCouponID) { (result) in
+                promise(result)
+            }
+        }
+
+        // Then
+        XCTAssert(result.isSuccess)
+        let coupon = try XCTUnwrap(result.get())
+        let expectedCoupon = sampleCoupon()
+        XCTAssertEqual(coupon, expectedCoupon)
```

Comment:
For the [CI failure](https://app.circleci.com/pipelines/github/woocommerce/woocommerce-ios/17091/workflows/c023536d-d19f-4608-98a8-7a3effca435c/jobs/32925), I think it's from `siteID` being `0` since it's an optional parameter in `Coupon`'s initializer.

When reviewing @pmusolino's PR earlier https://github.com/woocommerce/woocommerce-ios/pull/5994#discussion_r794189721, I was actually just wondering about the new way to set `siteID` in `Coupon` compared to the old way to decode `siteID` from the user info passed by the mapper ([example](https://github.com/woocommerce/woocommerce-ios/blob/d9e5292b52e53e04a8233abb48db5d2cb68516ac/Networking/Networking/Model/Product/Product.swift#L271-L273)). The new way saves some boilerplate to initialize from a decoder, but we have to remember to set the `siteID` correctly everywhere 🤔

Reply:
You are right, this is a trade-off between the two ways of decoding. I would say the new way should only be used in cases when manually writing the initializer with a decoder is too much troublesome, and we should be cautious of cases like this 😅. I've fixed the initializer for the sample coupon in this case in 0e989e2ed111c74da3413e9ea06be1bb02f4023f.